### ISSUE TITLE
Filter keys not being passed to the model in the correct format

### ISSUE THREAD

**Issue by chrisdpeters** [tiagopog/jsonapi-utils#39]
A simple example resource with filterable multi-word attribute:

```ruby
class PageResource < JSONAPI::Resource
  attribute :trashed_at
  filter :trashed_at
end
```

And a contrived spec that attempts to use the attribute:

```ruby
it 'returns an array containing the non-trashed page', :aggregate_failures do
  get('/pages', { filter: { 'trashed-at' => nil })
  expect(response).to be_ok
end
```

The response ends up including an error that indicates that the filter isn't being translated back to underscored before being passed to the model:

```json
{
  "errors": [
    {
      "title": "Internal Server Error",
      "detail": "Internal Server Error",
      "code": "500",
      "status": "500",
      "meta": {
        "exception": "PG::UndefinedColumn:    ERROR:  column pages.trashed-at does not exist ..."
      }
    }
  ]
}
```

I have these configurations in place:

```ruby
#:underscored_key, :camelized_key, :dasherized_key, or custom
config.json_key_format = :dasherized_key

#:underscored_route, :camelized_route, :dasherized_route, or custom
config.route_format = :dasherized_route
```

**Comment by tiagopog** [comment]
Weird but I will surely check. Thanks!

**Comment by gee-forr** [comment]
Have the same issue. Happy to help out with extra data if you need it.

**Comment by tiagopog** [comment]
Just checked that the reported bug is indeed happening. I may come back with a fix tomorrow (probably). Thanks for reporting :-)

**Comment by tiagopog** [comment]
`v0.5.0.beta4` has just released. I may work on a fix for `v0.4.x` later today.
